src/ng/q.js

```diff
@@ -354,8 +354,8 @@ function qFactory(nextTick, exceptionHandler) {
       var that = this;
       var done = false;
       try {
-        if ((isObject(val) || isFunction(val))) then = val && val.then;
-        if (isFunction(then)) {
+        if(isPromiseLike(val)) {
+          then = val && val.then
```

Comment:
I think fetching `then` must only be done once. So fetching it inside `isPromiseLike` and again here already breaks things: https://github.com/promises-aplus/promises-spec#notes